src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/Functions/SubscriptionManagerTests.cs

```diff
@@ -0,0 +1,554 @@
+    {
+        ReleaseAmendmentPublishedId = "release-amendment-published-id",
+        ReleaseAmendmentPublishedSupersededSubscribersId = "release-amendment-published-superseded-subscribers-id",
+        ReleasePublishedId = "release-published-id",
+        ReleasePublishedSupersededSubscribersId = "release-published-superseded-subscribers-id",
+        SubscriptionConfirmationId = "subscription-confirmation-id",
+        SubscriptionVerificationId = "subscription-verification-id"
+    };
+
+    [Fact]
+    public async Task SendsSubscriptionVerificationEmail()
+    {
+        // Arrange (mocks)
+        var storageTableService = new StorageTableService(Options.Create(new AppSettingOptions
+        {
+            TableStorageConnectionString = fixture.TableStorageConnectionString()
+        }));
+
+        var notificationClient = MockNotificationClient(HttpStatusCode.OK, "response content");
+        var notificationClientProvider = new Mock<INotificationClientProvider>(MockBehavior.Strict);
+        notificationClientProvider.Setup(mock =>
+                mock.Get())
+            .Returns(notificationClient);
+
+
+        var tokenService = new Mock<ITokenService>(MockBehavior.Strict);
+        tokenService.Setup(mock =>
+                mock.GenerateToken("test1@test.com", It.IsAny<DateTime>()))
+            .Returns("activation-code-1");
+
+        var emailService = new Mock<IEmailService>(MockBehavior.Strict);
+        emailService.Setup(mock =>
+            mock.SendEmail(notificationClient, "test1@test.com", "subscription-verification-id",
+                It.IsAny<Dictionary<string, dynamic>>()));
+
+        var notifierFunction = BuildFunction(
+            storageTableService: storageTableService,
+            notificationClientProvider: notificationClientProvider.Object,
+            tokenService: tokenService.Object,
+            emailService: emailService.Object);
+
+        var request = new NewPendingSubscriptionRequest
+        {
+            Id = "test-id-1",
+            Slug = "test-publication-slug-1",
+            Email = "test1@test.com",
+            Title = "Test Publication Title 1"
+        };
+
+        // Act
+        var result =
+            await notifierFunction.RequestPendingSubscriptionFunc(request, new TestFunctionContext(),
+                new CancellationToken());
+
+        // Assert
+        var okResult = Assert.IsType<OkObjectResult>(result);
+        Assert.NotNull(okResult);
+        Assert.Equal(200, okResult.StatusCode);
+
+        emailService.Verify(mock =>
+            mock.SendEmail(notificationClient, "test1@test.com", "subscription-verification-id",
+                It.Is<Dictionary<string, dynamic>>(d =>
+                    AssertEmailTemplateValues(d, "Test Publication Title 1",
+                        "https://localhost:3000/subscriptions/test-publication-slug-1/confirm-subscription/activation-code-1",
```

Comment:
Is this base URL/port guaranteed to be remain constant across dev machines/build pipelines/environments, or would a config variable be more robust here?

Reply:
It is actually is, because it's being fed in when the class is built (`Options.Create(new AppSettingOptions { PublicAppUrl = "https://localhost:3000" })` . 
I did think about making it a config variable, but wasn't sure if we already had those set in all environments and was being lazy :D I do agree it would be neater